### PR TITLE
fix: prevent jailed observers from voting

### PR DESCRIPTION
**Pull request by kingpinXD**
# Description
Closes https://github.com/zeta-chain/protocol-private/issues/287

- Verifies that the observers are not jailed when adding votes
- Verifies that the observers are bonded.
- Improves error propagation for the `IsValidObserver` function

# How Has This Been Tested?

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved observer validation to prevent jailed or tombstoned observers from voting or performing restricted actions.
  - Enhanced error handling and messaging for observer authorization failures, providing more specific reasons (e.g., jailed, tombstoned, not a validator).
- **Tests**
  - Updated and expanded test coverage for observer validation scenarios, including new error conditions and edge cases.
- **Documentation**
  - Updated changelog to reflect the fix preventing jailed observers from voting.
- **Refactor**
  - Standardized observer validation to use error-based checks for more descriptive feedback and consistent logic across the application.

### CODE REVIEW

File: x/observer/keeper/voting.go [hunk 1]
```diff
@@ -42,19 +43,34 @@
...
-	return true
+	consAddress, err := validator.GetConsAddr()
+	if err != nil {
+		return sdkerrors.Wrapf(types.ErrInvalidAddress, "invalid consensus address for observer: %s", address)
+	}
+	if k.slashingKeeper.IsTombstoned(ctx, consAddress) {
+		return sdkerrors.Wrapf(types.ErrValidatorTombstoned, "observer is tombstoned: %s", address)
+	}
+	return nil
 }
```
[the remark this answers is not in the document]
Comment (kingpinXD) [reply x/observer/keeper/voting.go:74]: address a test of Invalid Operator addres 
`not authorized if the observer address is invalid`

TestIsValidObserver_InvalidConsAddr , is not required as this is fetched from the SDK store , there is no way to set a validator which invalid consensus address

**Review by lumtis — COMMENTED** [review]
I would prefer to keep this change out of v31 for now as we're changing core logic of the system. I don't think it's a huge priority with the current assumption on the observer set.

File: x/observer/keeper/msg_server_vote_blame.go [hunk 2]
```diff
@@ -24,9 +24,9 @@ func (k msgServer) VoteBlame(
 		return nil, sdkerrors.Wrapf(cctypes.ErrUnsupportedChain, "%s, ChainID %d", voteBlameID, msg.ChainId)
 	}
 
-	if ok := k.IsNonTombstonedObserver(ctx, msg.Creator); !ok {
-		return nil, sdkerrors.Wrap(
-			types.ErrNotObserver, voteBlameID)
+	err := k.IsValidObserver(ctx, msg.Creator)
+	if err != nil {
+		return nil, err
```
[the remark this answers is not in the document]
Comment (kingpinXD) [reply x/observer/keeper/msg_server_vote_blame.go:29]: I think its fine since the errors retunred are alrady typed

**Review by lumtis — APPROVED** [review]
